File: 范例/Accpt/16xx/p1631.pas
PROGRAM p1631;

TYPE
	information=record
	hp,armor,cool,amount,power,now			:Longint;
	name							:String;
	end;

VAR
	king1,king2						:Information;
        s1,s2                                                   :Longint;

PROCEDURE Main;
var
        atack1,atack2                                           :Longint;
        time1,time2                                             :Longint;
begin
atack1:=(king2.power-king1.armor);
if atack1<1 then atack1:=1;
atack2:=(king1.power-king2.armor);
if atack2<1 then atack2:=1;

atack1:=atack1 * king2.amount;
atack2:=atack2 * king1.amount;

time1:=king1.hp div atack1;
if king1.hp mod atack1<>0 then inc(time1);
time1:=time1*king2.cool;
time2:=king2.hp div atack2;
if king2.hp mod atack1<>0 then inc(time2);
time2:=time2*king1.cool;

if time1<time2 then
        writeln(king2.name,' Survive!') else
if time1>time2 then
        writeln(king1.name,' Survive!') else
        writeln('All Die!');
end;

BEGIN
//assign(input,'p.in');
//reset(input);
while not eof do
begin
with king1 do
begin
readln(name);
readln(hp,armor,cool,amount,power);
s1:=hp;
now:=cool;
end;
with king2 do
begin
readln(name);
readln(hp,armor,cool,amount,power);
now:=cool;
s2:=hp;
end;
Main;
if not eof then readln;
end;
END.

File: Vol.07(1600-1699)/#1631 The King of the Survives/p1631.cpp
// 2002823 2009-09-26 10:59:08 Accepted  1631 C++ 0 184 呆滞的慢板 

// 星际争霸，水题，但题目极度有病，注意输入顺序是 name HP Armor Cool Amount Power ws 

#include <iostream>
#include <string>
using namespace std;

struct Unit {    
    string name;
    long long Armor, HP, Power, Amount, Cool;
    bool get() {
        if( cin >> name
                >> HP
                >> Armor
                >> Cool
                >> Amount
                >> Power
                >> ws
        ) return true;
        else return false;
    }
};

long long QTime(Unit a, Unit b) {
    long long DPS = a.Power - b.Armor;
    if(DPS < 1) DPS = 1;
    DPS *= a.Amount;
    int ans = b.HP / DPS;
    if(b.HP % DPS) ++ans;
    return a.Cool * ans;
}

int main() {
    Unit A, B;
    while(A.get() && B.get()) {
        long long t1 = QTime(A, B);
        long long t2 = QTime(B, A);
        if(t1 > t2) cout << B.name << " Survive!\n";
        else if(t1 < t2) cout << A.name << " Survive!\n";
        else cout << "All Die!" << endl;
    }
}
